lib/ansible/modules/cloud/openstack/os_coe_cluster.py

```diff
@@ -0,0 +1,282 @@
+#!/usr/bin/python
+
+# Copyright (c) 2018 Catalyst IT Ltd.
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: os_coe_cluster
+short_description: Add/Remove COE cluster from OpenStack Cloud
+extends_documentation_fragment: openstack
+version_added: "2.7"
+author: "Feilong Wang (@flwang)"
+description:
+   - Add or Remove COE cluster from the OpenStack Container Infra service.
+options:
+   availability_zone:
+     description:
+       - Ignored. Present for backwards compatibility
+   cluster_template_id:
+     description:
+        - The template ID of cluster template.
+     required: true
+   discovery_url:
+       description:
+         - Url used for cluster node discovery
+   docker_volume_size:
+      description:
+         - The size in GB of the docker volume
+   flavor_id:
+      description:
+         - The flavor of the minion node for this ClusterTemplate
+   keypair_id:
```

Comment:
I'd change this to keypair if it can be a name too.

Reply:
Nice catch. Unfortunately, Magnum is using inconsistent attribute name for keypair in `cluster` and `cluster template`. In OpenStack, there is no id for keypair, just the name. So I'm going to rename it to 'keypair'. And may propose another patch to improve the cluster template later. Thanks.